ucm2/USB-Audio/Behringer/UMC204HD-HiFi.conf

```diff
@@ -0,0 +1,120 @@
+LibraryConfig.pcm.Config {
+	pcm.umc204hd_stereo_out {
+		@args [ CARD CHN0 CHN1 ]
+		@args {
+			CARD.type string
+			CHN0.type integer
+			CHN1.type integer
+		}
+		type dshare
+		ipc_key 572442
+		ipc_perm 0600
+		slave {
+			pcm {
+				type hw
+				card $CARD
+				device 0
+			}
+			channels 4
+		}
+		bindings.0 $CHN0
+		bindings.1 $CHN1
+	}
+
+	pcm.umc204hd_mono_in {
+		@args [ CARD CHN0 CHN1 ]
+		@args {
+			CARD.type string
+			CHN0.type integer
+			CHN1.type integer
+		}
+		type dsnoop
+		ipc_key 572542
+		ipc_perm 0600
+		slave {
+			pcm {
+				type hw
+				card $CARD
+				device 0
+			}
+			channels 2
+		}
+		bindings.0 $CHN0
+		bindings.1 $CHN1
+	}
+
+	pcm.umc204hd_line1 {
+		@args [ CARD ]
+		@args.CARD.type string
+		type empty
+		slave.pcm {
+			@func concat
+			strings [ "umc204hd_stereo_out:" $CARD ",0,1" ]
+		}
+	}
+
+	pcm.umc204hd_line2 {
+		@args [ CARD ]
+		@args.CARD.type string
+		type empty
+		slave.pcm {
+			@func concat
+			strings [ "umc204hd_stereo_out:" $CARD ",2,3" ]
+		}
+	}
+
+	pcm.umc204hd_mic1 {
+		@args [ CARD ]
+		@args.CARD.type string
+		type empty
+		slave.pcm {
+			@func concat
+			strings [ "umc204hd_mono_in:" $CARD ",0,0" ]
```

Comment:
There has to be a better way. This card has two *mono* inputs, but I couldn't get them working as mono, so instead this maps them as stereo where both channels have the same audio... Any idea how to get them properly mapped as mono inputs?

Reply:
Change ",0,0" to ",0" only with my recommendations. The `bindings` lines are the key - https://www.alsa-project.org/alsa-doc/alsa-lib/pcm_plugins.html .

According (yours?) alsa-info output (https://bugzilla-attachments.redhat.com/attachment.cgi?id=1845960), there are volume controls. You should describe them using `PlaybackMixerElem` and `CaptureMixerElem` values.